frontend/src/page/Login.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { signInWithPopup } from 'firebase/auth';
+import './loginsignup.css';
+import googlelogo from '../assets/google_logo.svg';
+import { auth, provider } from '../services/firebase';
+import { getUser } from '../api/user';
+import Form from '../components/Form';
+
+function Login() {
+  const [userName, setUserName] = useState('');
+  const navigate = useNavigate();
+  const [username, hasError] = useState(false);
+  const googleLogin = async () => {
+    signInWithPopup(auth, provider);
+    const res = await getUser(userName);
+    if (res.data === userName) {
+      navigate('/');
+    } else {
+      hasError(!username);
```

Comment:
`hasError` should be the boolean state variable so this should be `setHasError` instead, follow the naming convention:
https://react.dev/reference/react/useState